### PR TITLE
Docstrings for tools.build and tools.file

### PR DESCRIPTION
**Pull request by lasote**
With https://github.com/conan-io/docs/pull/2531

Minor refactors

### CODE REVIEW

File: conan/tools/build/cross_building.py [hunk 1]
```diff
@@ -1,7 +1,19 @@
 
 def cross_building(conanfile=None, skip_x64_x86=False):
+    """
+    Check it we are cross building comparing the *build* and *host* settings. Returns ``True``
```
Comment (czoido) [inline conan/tools/build/cross_building.py]:
```suggestion
    Check if we are cross building comparing the *build* and *host* settings. Returns ``True``
```

File: conan/tools/files/conandata.py [hunk 2]
```diff
@@ -8,9 +8,11 @@
 
 def update_conandata(conanfile, data):
     """
-    this only works for updating the conandata on the export() method, it seems it would
-    be plain wrong to try to change it anywhere else
+
```
Comment (czoido) [inline conan/tools/files/conandata.py:17]: Maybe a brief explanation of what the tool does here?

**Review by czoido — APPROVED** [review]

File: conan/tools/build/cpu.py [hunk 3]
```diff
@@ -5,6 +5,13 @@
 
 
 def build_jobs(conanfile):
+    """
+    Get the number of jobs to use while building. Read from the conf ``tools.build:jobs``
+    or autodetected
```
Comment (memsharded) [inline conan/tools/build/cpu.py]:
```suggestion
    or autodetected, by default it will return the number of CPUs available.
```

**Review by memsharded — APPROVED** [review]
good job

**Pull request merged**